From 7da9dd918621e1ccb77ce62f1a4133d8416a8be7 Mon Sep 17 00:00:00 2001
From: Alexander Sack <asac@pantacor.com>
Date: Fri, 9 May 2025 09:09:10 +0000
Subject: [PATCH] Introduce option to inject distro version and name into
 pantavisor version.c

* support PANTAVISOR_DISTRO_VERSION define for cmake
* support PANTAVISOR_DISTRO_NAME define for cmake
---
 CMakeLists.txt | 4 +++-
 gen_version.sh | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 086f59b..bf914c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,8 @@ set(PANTAVISOR_DEBUG OFF CACHE BOOL "Enable debug features for Pantavisor")
 set(PANTAVISOR_APPENGINE OFF CACHE BOOL "Build Pantavisor as appengine")
 set(PANTAVISOR_DEFAULTS_SKIP_INSTALL OFF CACHE BOOL "Skip installing defaults/ to etc/pantavisor")
 set(PANTAVISOR_PVS_SKIP_INSTALL OFF CACHE BOOL "Skip installing developer trust store")
+set(PANTAVISOR_DISTRO_VERSION "" CACHE STRING "Set a distro version to include in pantavisor build version")
+set(PANTAVISOR_DISTRO_NAME "" CACHE STRING "Set a distro name to include in pantavisor build version")
 
 # set cmake defintions
 set(CODE_INSTALL_BASE "\$ENV{DESTDIR}")
@@ -44,7 +46,7 @@ add_custom_command(
     OUTPUT  version.c
     COMMAND sh
 	-c
-	\"cd ${CMAKE_CURRENT_SOURCE_DIR}\; ./gen_version.sh CMAKE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SYSTEM_PROCESSOR}\"
+	\"cd ${CMAKE_CURRENT_SOURCE_DIR}\; DISTRO_NAME=${PANTAVISOR_DISTRO_NAME} DISTRO_VERSION=${PANTAVISOR_DISTRO_VERSION} ./gen_version.sh CMAKE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SYSTEM_PROCESSOR}\"
 )
 
 # Build and Install Pantavisor
diff --git a/gen_version.sh b/gen_version.sh
index 96f3b0e..9fd7b1b 100755
--- a/gen_version.sh
+++ b/gen_version.sh
@@ -10,7 +10,7 @@ if [ "$type" = CMAKE ]; then
     gitdescribe=$(git describe --tags)
 
     echo "const char *pv_build_manifest = \"\";" > $buildir/version.c
-    echo "const char *pv_build_version = \"${gitdescribe}-${now}\";" >> $buildir/version.c
+    echo "const char *pv_build_version = \"${gitdescribe}-${now}${DISTRO_NAME:+ | ${DISTRO_NAME}}${DISTRO_VERSION:+ (${DISTRO_VERSION})}\";" >> $buildir/version.c
     echo "const char *pv_build_date = \"${now}\";" >> $buildir/version.c
     echo "const char *pv_build_arch = \"${3}\";" >> $buildir/version.c
 
